### ISSUE TITLE
[BUG] - You don't seem connected to the internet

### ISSUE THREAD

**Issue by HifuminG** [bottlesdevs/Bottles#181]
**Describe the bug**
I just installed Bottles from the AUR and for some reason there is an icon in the upper right corner that says that I "don't seem connected to the internet" but I am. I tried both wifi and wired and no matter how many times I click that icon, it stays the same.

**Installation**
- Method: AUR
- Version: n/a

**To Reproduce**
n/a

**Expected behavior**
n/a

**Screenshots**
n/a

**Desktop**
- OS: Arch linux
- Version: n/a

**Additional context**
n/a

**Comment by mirkobrombin** [comment]
Please fill a bug report next time. I fixed the layout of your message.

I need the following informations:
- Which package did you install from AUR
- Can you reach 1.1.1.1? 
- Can you run Bottles from the terminal to get any logs, exceptions?

**Comment by HifuminG** [comment]
- I installed [this one](https://aur.archlinux.org/packages/bottles) with all its dependencies.
- I'm not sure how to do that, but just typing it in my web browser, I can't connect to it. I can ping to it though.
- Running it from the terminal returns:
`INFO: Connection status: offline ...`
`INFO: Connection status: offline ...`
`INFO: Connection status: offline ...`
And every time I click on the "you don't seem to be connected to the internet" icon, the same line appears on the terminal.

I'm sorry about the layout, I didn't know whether or not this was an actual bug or just me doing something I shouldn't

**Comment by mirkobrombin** [comment]
> * I installed [this one](https://aur.archlinux.org/packages/bottles) with all its dependencies.
> * I'm not sure how to do that, but just typing it in my web browser, I can't connect to it. I can ping to it though.
> * Running it from the terminal returns:
>   `INFO: Connection status: offline ...`
>   `INFO: Connection status: offline ...`
>   `INFO: Connection status: offline ...`
>   And every time I click on the "you don't seem to be connected to the internet" icon, the same line appears on the terminal.
> 
> I'm sorry about the layout, I didn't know whether or not this was an actual bug or just me doing something I shouldn't

Just type in your terminal:
```
ping 1.1.1.1
```

**Comment by HifuminG** [comment]
> Just type in your terminal:
> 
> ```
> ping 1.1.1.1
> ```

As I said, I can ping to it with no issues.
`--- 1.1.1.1 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2003ms
rtt min/avg/max/mdev = 17.584/29.236/51.188/15.531 ms`

**Comment by mirkobrombin** [comment]
> > Just type in your terminal:
> > ```
> > ping 1.1.1.1
> > ```
> 
> As I said, I can ping to it with no issues.
> `--- 1.1.1.1 ping statistics --- 3 packets transmitted, 3 received, 0% packet loss, time 2003ms rtt min/avg/max/mdev = 17.584/29.236/51.188/15.531 ms`

I had missed that part of the message. I investigate as soon as possible.

**Comment by guihkx** [comment]
I'm having the same issue, I think.

**Describe the bug**
I'm running Bottles via Flatpak and it says I'm offline as soon as I launch it.

**Installation**
 - Method: Flatpak
 - Version: 3.1.3

**To Reproduce**
Steps to reproduce the behavior:
1. Launch Bottles using a terminal:

     ```
    $ flatpak run com.usebottles.bottles
     ```
2. Observe output:

    ```
    Gtk-Message: 00:14:44.882: Failed to load module "canberra-gtk-module"
    2021-04-08 00:14:45 INFO: Connection status: offline …
    2021-04-08 00:14:45 INFO: Connection status: offline …
    2021-04-08 00:14:45 INFO: Connection status: offline …
    2021-04-08 00:14:45 INFO: Bottles Started!
    2021-04-08 00:14:47 WARNING: No dxvk found.
    2021-04-08 00:14:47 INFO: Connection status: offline …
    2021-04-08 00:14:47 WARNING: No runners found.
    2021-04-08 00:14:47 INFO: Connection status: offline …
    2021-04-08 00:14:47 INFO: Connection status: offline …
    2021-04-08 00:14:47 INFO: Connection status: offline …
    ```

**Expected behavior**
Bottles to have a working internet connection.

**Screenshots**
![image](https://user-images.githubusercontent.com/626206/113963161-80e6dc00-97ff-11eb-99e9-92af787d34dc.png)

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Version: rolling release

**Additional context**
I do have a working internet connection, and the code that checks for that on Bottles seems to be this one:

https://github.com/bottlesdevs/Bottles/blob/81d96f06a569b1dfa837be03c6dea7b9bf3431ed/src/utils.py#L41-L62

If I understand correctly, it attempts to resolve the `usebottles.com` domain, but it fails for some reason.

I can resolve it manually just fine, though (and other Flatpak programs don't have a similar issue):

```
$ nslookup usebottles.com
Server:		127.0.0.1
Address:	127.0.0.1#53

Non-authoritative answer:
Name:	usebottles.com
Address: 104.21.92.184
Name:	usebottles.com
Address: 172.67.197.25
Name:	usebottles.com
Address: 2606:4700:3037::ac43:c519
Name:	usebottles.com
Address: 2606:4700:3034::6815:5cb8
```

**Comment by guihkx** [comment]
Hmm, at least for me this issue is not exclusive to the Flatpak build. I tried with the official AppImage build and I have the exact same output:

![image](https://user-images.githubusercontent.com/626206/113964211-6b72b180-9801-11eb-9e9d-f9f5202233a6.png)

**Comment by guihkx** [comment]
Oooh, I get it now.

Because I was testing version 3.1.3 (on both Flatpak and AppImage), the code that I linked was not from version 3.1.3, which used to be this one:

https://github.com/bottlesdevs/Bottles/blob/a03fe924d5ff4b35ab486661701948c4517a53fc/src/utils.py#L41-L62

And indeed, my ISP is the one blocking requests to `1.1.1.1` on ports 80, 443 and 53 (and maybe others, but I can `ping 1.1.1.1` just fine).

Now I'll build a Flatpak image locally on a more recent commit and give it a shot.

**Comment by guihkx** [comment]
Yep, Flatpak version works now on commit 0aaad344bfa46e0272ac6624934fb1d291e888bb:

![image](https://user-images.githubusercontent.com/626206/113969187-e5f3ff00-980a-11eb-8a0d-9026657ea80c.png)

Sorry for the spam, guys! 😬

**Comment by mirkobrombin** [comment]
@guihkx @HifuminG the fix is ​​already there (as you could see from your tests), I just haven't had time to publish a new release yet. Thanks for testing, now I know it works, I will publish the release soon or in the day.

**Comment by mirkobrombin** [comment]
Fixed in 3.1.4
